plugins/woocommerce/includes/class-wc-cart.php

```diff
@@ -1511,8 +1511,8 @@ public function calculate_shipping() {
 			}
 		}
 
-		$this->set_shipping_total( array_sum( $shipping_costs ) );
-		$this->set_shipping_tax( array_sum( $merged_taxes ) );
+		$this->set_shipping_total( array_sum( array_filter( $shipping_costs ) ) );
+		$this->set_shipping_tax( array_sum( array_filter( $merged_taxes ) ) );
```

Comment:
https://github.com/woocommerce/woocommerce/issues/58064

Reply:
Hey @kalessil - Is this planned for WC 9.9 ?

Reply:
@PanosSynetos : don't know yet - primary reason for the change is to get the tests passing and connect dots to any existing issues. Now I need to coordinate with the teams on how to proceed with my fixes, so if you have any inputs or want to pull anyone in on the conversation, that would be helpful!

Reply:
This issue is causing Conditional Shipping and Payments unit tests to fail.

Let me tag @mikejolley too, as he discussed with @jimjasson over slack on Friday

What is important to note here, is that this doesn't cause a warning with PHP8.5 and WC 9.8.5, and all @mikejolley did, was to move a line further up in this PR.
- https://github.com/woocommerce/woocommerce/pull/55804

So, something else must have changed elsewhere, and I wonder if by doing an `array_filter` we mask/hide another issue.

Reply:
> wondering if something is returning an empty string here

From our tests, we noticed that Local Pickup methods that don't have a price have an empty string as the `cost`.